admin/refresh_spdx/add_licenses.py

```diff
@@ -0,0 +1,83 @@
+"""Refresh a BUILD file of license_kinds with new ones from licenses.json.
+
+Usage:
+  wget https://github.com/spdx/license-list-data/raw/master/json/licenses.json
+  python add_licenses.py >BUILD.more
+  cat BUILD.more ../../licenses/spdx/BUILD
+
+TODO(aiuto): Refine this enough to update BUILD in place.
+TODO(aiuto): Refer to it from /licenses/spdx/BUILD
+"""
+
+import json
+from string import Template
+import re
+
+def load_json(path):
+  """Loads a JSON file and returns the data.
+
+  Args:
+    path: (str) a file path.
+  Returns:
+    (dict) the parsed data.
+  """
+  with open(path, 'r') as fp:
+    ret = json.load(fp)
+  return ret
+
+
+def gather_names(path):
+  name_match = re.compile(r'^ *name *= *"([^"]*)"')
+  ret = []
+  with open(path, 'r') as fp:
+    for line in fp:
+      m = name_match.match(line)
+      if m:
+        ret.append(m.group(1))
+  return ret
+
+"""
+{
+  "licenseListVersion": "3.8-57-gca4f142",
+  "licenses": [
+    {
+      "reference": "./0BSD.html",
+      "isDeprecatedLicenseId": false,
+      "detailsUrl": "http://spdx.org/licenses/0BSD.json",
+      "referenceNumber": "232",
+      "name": "BSD Zero Clause License",
+      "licenseId": "0BSD",
+      "seeAlso": [
+        "http://landley.net/toybox/license.html"
+      ],
+      "isOsiApproved": true
+    },
+"""
+
+
+def add_new(already_declared, licenses):
+  template = Template((
```

Comment:
This is fine, but why not just let the json library format a dict for you?

Reply:
I don't quite understand the suggestion. I need to emit bazel style rules, not dicts, and I don't know of any templating language within the json library

Reply:
Ignore my comment, I misread the CL.